packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -158,6 +158,90 @@ class _SnackBarActionState extends State<SnackBarAction> {
 /// A SnackBar with an action will not time out when TalkBack or VoiceOver are
 /// enabled. This is controlled by [AccessibilityFeatures.accessibleNavigation].
 ///
+/// {@tool dartpad --template=stateless_widget_scaffold_center}
+///
+/// Here is an example of a [SnackBar] with an [action] button implemented using
+/// [SnackBarAction].
+///
+/// ```dart preamble
+/// class _ShowSnackBarButton extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return ElevatedButton(
+///       child: Text("Show Snackbar"),
+///       onPressed: () {
+///         Scaffold.of(context).showSnackBar(
```

Comment:
Use ScaffoldMessenger instead of Scaffold to call showSnackBar function.

Reply:
^ This is causing the analyzer check failure that we're seeing. I still haven't tried ScaffoldMessenger yet myself, so it'll be interesting to see the new code :)

Reply:
Also, the docs are at master-api.flutter.dev: https://master-api.flutter.dev/flutter/material/ScaffoldMessenger-class.html

Since ScaffoldMessenger hasn't yet made it into `stable` yet.

Reply:
I changed the code to use ScaffoldMessenger to show the SnackBar. Please review.